test_conformance/basic/test_intmath.cpp

```diff
@@ -0,0 +1,233 @@
+        inputC[i] = (T)genrand_int64(d);
+    }
+
+    size_t datasize = sizeof(T) * num_elements * N;
+
+    // Create device buffers.
+    for (int i = 0; i < ARRAY_SIZE(streams); i++)
+    {
+        streams[i] =
+            clCreateBuffer(context, CL_MEM_READ_WRITE, datasize, NULL, &err);
+        test_error(err, "clCreateBuffer failed");
+    }
+
+    // Copy input data to device.
+    err = clEnqueueWriteBuffer(queue, streams[0], CL_TRUE, 0, datasize,
+                               inputA.data(), 0, NULL, NULL);
+    test_error(err, "clEnqueueWriteBuffer failed\n");
+    err = clEnqueueWriteBuffer(queue, streams[1], CL_TRUE, 0, datasize,
+                               inputB.data(), 0, NULL, NULL);
+    test_error(err, "clEnqueueWriteBuffer failed\n");
+    err = clEnqueueWriteBuffer(queue, streams[2], CL_TRUE, 0, datasize,
+                               inputC.data(), 0, NULL, NULL);
+    test_error(err, "clEnqueueWriteBuffer failed\n");
+
+    std::string build_options = "-DTYPE=";
+    build_options += typestr;
+
+    // Run test for each operation
+    for (auto test : tests)
+    {
+        log_info("%s... ", test.name);
+
+        // Create kernel and set args
+        clProgramWrapper program;
+        clKernelWrapper kernel;
+        err = create_single_kernel_helper(context, &program, &kernel, 1,
+                                          &test.kernel_code, test.name,
+                                          build_options.c_str());
+        test_error(err, "create_single_kernel_helper failed\n");
+
+        err = clSetKernelArg(kernel, 0, sizeof(cl_mem), &streams[0]);
+        err |= clSetKernelArg(kernel, 1, sizeof(cl_mem), &streams[1]);
+        err |= clSetKernelArg(kernel, 2, sizeof(cl_mem), &streams[2]);
+        err |= clSetKernelArg(kernel, 3, sizeof(cl_mem), &streams[3]);
+        test_error(err, "clSetKernelArgs failed\n");
+
+        // Run kernel
+        size_t threads[1] = { static_cast<size_t>(num_elements) };
+        err = clEnqueueNDRangeKernel(queue, kernel, 1, NULL, threads, NULL, 0,
+                                     NULL, NULL);
+        test_error(err, "clEnqueueNDRangeKernel failed\n");
+
+        // Read results
+        err = clEnqueueReadBuffer(queue, streams[3], CL_TRUE, 0, datasize,
+                                  output.data(), 0, NULL, NULL);
+        test_error(err, "clEnqueueReadBuffer failed\n");
+
+        // Verify results
+        for (int i = 0; i < num_elements * N; i++)
+        {
+            T r = test.ref(inputA[i], inputB[i], inputC[i]);
+            if (r != output[i])
+            {
+                log_error("failed\n");
+                return -1;
+            }
+        }
+        log_info("passed\n");
+    }
+
+    return err;
+}
+
+int test_intmath_int(cl_device_id device, cl_context context,
+                     cl_command_queue queue, int num_elements)
+{
+    return test_intmath<cl_uint, 1>(device, context, queue, num_elements,
```

Comment:
Previously cl_int types was tested not cl_uint. Maybe both should be tested? The same remark to all other cases

Reply:
This depends on whether we want these tests to be more than just a basic smoke test, as discussed above. I switched to `uint` as there was a signed overflow bug in the previous tests. I can add signed `int` back if we really want that coverage here (just need to do some range reduction on the input values to avoid overflow).

Reply:
I am ok with this but then little confusion with function names (test_intmath_int2).  Consider changing names to test_intmath_uint2?

Reply:
I think we shouldn't do that in this PR but as part of the follow up issue where we look into folding these tests into integer_ops.

Reply:
then LGTM